sys/linux/bpf.txt

```diff
@@ -400,19 +408,25 @@ bpf_obj_get {
 	file_flags	flags[bpf_open_flags, int32]
 }
 
+bpf_attach_targets [
+	zero	const[0, int32]
+	cgroup	fd_cgroup
+	map	fd_bpf_map
+] [size[32]]
+
 bpf_attach_arg {
-	target_fd	fd_cgroup
+	target_fd	bpf_attach_targets
 	attach_bpf_fd	fd_bpf_prog
 	type		flags[bpf_attach_type, int32]
 	flags		flags[bpf_attach_flags, int32]
 	replace_bpf_fd	fd_bpf_prog
 }
 
 bpf_detach_arg {
-	target	const[0, int32]
+	target	bpf_attach_targets
 	prog	fd_bpf_prog
 	type	flags[bpf_attach_type, int32]
-	flags	const[0, int32]
+	flags	flags[bpf_attach_flags, int32]
```

Comment:
This is detach arg, do we still need bpf_attach_targets and flags here? They may be unused here. But I did not double check now.

Reply:
Sorry, you're right. Flags are actually unused for detach, the last valid field is `attach_type`.